### ISSUE TITLE
Enable building and publishing of portable PDBs for x-plat libraries

### ISSUE THREAD

**Issue by weshaggard** [dotnet/corefx#5733]
For our x-plat builds we need to start producing portable PDBs. Once we do that we need to also ensure they are getting published like our windows PDBs.

**Comment by joperezr** [comment]
Waiting for Microsoft/msbuild#459 to get fixed in order to be able to enable this.

**Comment by joperezr** [comment]
Closing since this was enabled with #5911

**Comment by weshaggard** [comment]
We need to keep this open to track the publishing of these symbols as well. Assigning to @markwilkie to track someone is correctly publishing these x-plat pdbs.

**Comment by joperezr** [comment]
Sorry, I thought a new issue would track the publishing of the symbols. I suppose that first we will have to close #5825 since we need to make sure that the PDBs we are generating now are valid before publishing them. I'll get to that issue next week.

**Comment by markwilkie** [comment]
Davis has been making great progress on symbols overall, as well as x-plat is specifics.  @dagood, could you update the issue w/ the latest for x-plat symbols please?

**Comment by dagood** [comment]
Sure. This is new terminology for me, though... the categories I've known so far are managed symbols, Windows native symbols, and xplat (non-Windows) native symbols. A few questions:
* Would it be fair to call these "xplat managed symbols"?
* Is there a good place to read up on what portable pdbs are for?
* Do we create portable pdbs on non-Windows platforms or everywhere?

**Comment by weshaggard** [comment]
https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/specs/PortablePdb-Metadata.md

Portable PDB's are a new x-plat compatible type of symbols that don't require Windows COM support. They are primarily used for x-plat but could also be used on windows as well. At this point they are still more limited in support so on windows we still use the existing PDB format but I expect overtime we will likely shift to using them everywhere.

**Comment by dagood** [comment]
Ah, thanks. When would a Portable PDB be published, then? If I understand correctly, managed code can be built on any platform, and we currently build them on Windows. Are we planning on publishing managed libraries from another platform?

**Comment by joperezr** [comment]
I'm not sure if I understood your question correctly, but essentially yes we do publish managed libraries from all platforms. The problem with this issue is that currently the portable PDBs that we generate x-plat are invalid and therefore can't be used to debug. The reason for it, is that since after building each library (and it's portable PDB) we run a tool in buildtools called GenFacades that adds content to the library and the symbols, and since GenFacades doesn't support Portable PDBs yet, then the ones we are left with are invalid.

That tool needs to first add support for Portable PDBs, and once that is done, we can start publishing those symbols as well.

**Comment by dagood** [comment]
Ah, thanks, I think I see my confusion now: there's a gap between the TFS build and the build pipeline that I hadn't seen yet. The TFS build does collect managed artifacts from xplat to use for managed runtime dlls, but the build pipeline just builds all the managed stuff on Windows. (In fact, xplat managed builds aren't even run by the build pipeline--only natives.) Since I've been working on the build pipeline I assumed that was the way things were done for managed libraries in TFS builds too.

@jhendrixMSFT Is this difference correct? Are there plans to build managed libs on each platform and combine them into managed package runtime dirs, or will we keep building them all at once?

Funnily, because these are the managed packages we're talking about, we're actually already publishing the portable PDBs. For example, check the "SYMBOLS" tab on https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.Diagnostics.Process. It's also on the internal MS symbol server.

Should I stop publishing managed symbols until GenFacades adds support?

**Comment by weshaggard** [comment]
>@jhendrixMSFT Is this difference correct? Are there plans to build managed libs on each platform and combine them into managed package runtime dirs, or will we keep building them all at once?

We were planning to do that originally but we had to change direction on that when we decided to put all these assets into the same package (aka FAT packages). So we are building all the managed assemblies as part of the same windows build right now and only building native binaries on the specific x-plat builds. 

Thinking about a little more I don't think there is anything more left to do with this work item now. As we do build portable PDB's now on x-plat but there isn't any additional work to publish symbols from there right now. The only symbols we need to publish from the x-plat builds are the native symbols at this time. If we ever do decide to ship managed built binaries from the x-plat builds we can revisit this but I suspect that will end up being the same as the windows builds.

I'm closing this @dagood but I would like to to at least verify that the portable PDB's we are building for our x-plat managed libraries on windows don't cause any issues with our symbol publishing plans.

**Comment by weshaggard** [comment]
@joperezr correct me if I'm wrong but the genfacades issues is tracked by a different issue correct?

**Comment by joperezr** [comment]
Looks like we never actually logged an issue for that problem. We used to have this one #5825 but that one was only to test if the portable PDBs could be used for debugging, so I closed it once the investigation was done but I didn't log a new one for fixing GenFacades. I'll do that now in BuildTools repo, since I think it makes more sense to have the issue logged there.

**Comment by dagood** [comment]
I see, thanks for going through the reasoning, Wes. I've only exercised the feeds from Windows so it's possible the symbol servers handle the portable PDBs incorrectly. I subscribed to https://github.com/dotnet/buildtools/issues/705 and I'm thinking that as part of finishing off that work we can make sure portable PDBs behave the same after being put through symbol servers.
